data/abilities.js

```diff
@@ -258,7 +256,11 @@ let BattleAbilities = {
 		desc: "If this Pokemon is a Greninja, it transforms into Ash-Greninja after knocking out a Pokemon. As Ash-Greninja, its Water Shuriken has 20 base power and always hits 3 times.",
 		shortDesc: "After KOing a Pokemon: becomes Ash-Greninja, Water Shuriken: 20 power, hits 3x.",
 		onSourceFaint(target, source, effect) {
-			if (effect && effect.effectType === 'Move' && source.template.speciesid === 'greninja' && source.hp && !source.transformed && source.side.foe.pokemonLeft) {
+			let remainingFoes = 0;
+			for (const side of this.sides) {
+				remainingFoes += side.n % 2 !== source.side.n % 2 ? side.pokemonLeft : 0;
```

Comment:
Originally I would have just written this as `if (side.n % 2 !== source.side.n % 2) remainingFoes += side.pokemonLeft;`, but looking at the rest of the PR I notice you use `!target.foes(true).some(poke => poke && !!poke.side.pokemonLeft)` later, maybe that could replace this whole loop?

Reply:
Oh, duh. Dunno why I did this one way.